### DISCUSSION TITLE
How to get a header from a response?

### DISCUSSION THREAD

**Discussion by vholik** [urql-graphql/urql#3475] [Q&A]
I am implementing session strategy and the problem is that I need to set a session token in localstorage. How to get a header from a client response?

I have tried to make a fetch wrapper but it seems like it's not a proper way to do this.

```
export const fetchPolyfill = (...args: any[]) =>
  // @ts-ignore
  fetch(...args).then((response) => {
    const newSession = response.headers.get("woocommerce-session");
    const existingSession = !isServer
      ? localStorage.getItem("woo-session")
      : null;

    if (!isServer && newSession && !existingSession) {
      localStorage.setItem("woo-session", newSession);
    }

    return response;
  });

```

Thanks in advance

**Comment by kitten** [comment]
> Note: See #3474 which basically contains the same question, since it was asked there too.

There's also other threads in discussions and issues on this topic which can be found with keywords like “response” and “headers”, e.g. https://github.com/urql-graphql/urql/discussions/1693
